pkg/sql/catalog/tabledesc/validate.go

```diff
@@ -865,15 +849,20 @@ func ValidateNotVisibleIndex(
 		return notices
 	}
 
-	notice := tableDesc.ForeachOutboundFK(func(fk *descpb.ForeignKeyConstraint) error {
-		// Case 2: The given index is an index on a child table that may be useful
-		// for FK check.
-		if index.IsHelpfulOriginIndex(fk.OriginColumnIDs) {
-			return notices
-		}
+	if index.IsPartial() || index.NumKeyColumns() == 0 {
```

Comment:
`index.NumKeyColumns() == 0` can legitimately happen for the primary index of a virtual table. Of course, such tables don't have primary indexes at all, so I'm not sure how much I like the fact that we pretend that we do, but this is not the time or place to deal with this.